Error Trying to Compile on MacOS (x86_64)

My ask is so cursed that I didn't want to event submit it but figured I should.

```
error[E0560]: struct `sigaction` has no field named `sa_restorer`
   --> src/main.rs:235:13
    |
235 |             sa_restorer: None,
    |             ^^^^^^^^^^^ `sigaction` does not have this field
    |
    = note: all struct fields are already assigned

error[E0560]: struct `sigaction` has no field named `sa_restorer`
   --> src/key_handlers.rs:537:17
    |
537 |                 sa_restorer: None,
    |                 ^^^^^^^^^^^ `sigaction` does not have this field
    |
    = note: all struct fields are already assigned

For more information about this error, try `rustc --explain E0560`.
error: could not compile `rwm` (bin "rwm") due to 2 previous errors
error: failed to compile `rwm v0.1.0 (/Users/kojinglick/Downloads/rwm-1.0.1)`, intermediate artifacts can be found at `/Users/xxx/Downloads/rwm-1.0.1/target`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -7,19 +7,28 @@ on:
 name: test
 jobs:
   required:
-    runs-on: ubuntu-latest
-    name: ubuntu / ${{ matrix.toolchain }}
+    runs-on: ${{ matrix.os }}
+    name: ${{ matrix.os }} / ${{ matrix.toolchain }}
     strategy:
       matrix:
         toolchain: [nightly]
+        os: [ubuntu-latest, macos-13]
+        include:
+          - os: macos-13
+            is_mac: 1
     steps:
       - uses: actions/checkout@v4
         with:
           submodules: true
-      - name: Install X libraries
+      - name: Install X libraries (ubuntu)
+        if: ${{ ! matrix.is_mac }}
         run: |
           sudo apt-get update
           sudo apt-get install libxft-dev libxinerama-dev
+      - name: Install X libraries (mac)
+        if: ${{ matrix.is_mac }}
+        run: |
+          brew install libxft libxinerama
       - name: Install ${{ matrix.toolchain }}
         uses: dtolnay/rust-toolchain@master
         with:

--- a/src/key_handlers.rs
+++ b/src/key_handlers.rs
@@ -534,6 +534,7 @@ pub(crate) unsafe extern "C" fn spawn(arg: *const Arg) {
                 // MaybeUninit the whole sigaction if I can avoid it
                 sa_mask: std::mem::zeroed(),
                 sa_flags: 0,
+                #[cfg(not(target_os = "macos"))]
                 sa_restorer: None,
             };
             libc::sigemptyset(&mut sa.sa_mask);

--- a/src/main.rs
+++ b/src/main.rs
@@ -232,6 +232,7 @@ fn setup() {
             sa_flags: libc::SA_NOCLDSTOP
                 | libc::SA_NOCLDWAIT
                 | libc::SA_RESTART,
+            #[cfg(not(target_os = "macos"))]
             sa_restorer: None,
         };
         libc::sigemptyset(&mut sa.sa_mask);

